### ISSUE TITLE
Changing the mac address doesn't work

### ISSUE THREAD

**Issue by brokenpylons** [dmacvicar/terraform-provider-libvirt#727]
### Linux distribution

```Arch ```

### Terraform version

Terraform v0.12.20

### Provider and libvirt versions

./terraform-provider-libvirt db587ad14bfb29aa9cbd56482c79f414e2f26409-dirty
Compiled against library: libvirt 5.10.0
Using library: libvirt 5.10.0
Running hypervisor: QEMU 4.2.0
Running against daemon: 5.10.0
___

## Checklist

- [ ] Is your issue/contribution related with enabling some setting/option exposed by libvirt that the plugin does not yet support, or requires changing/extending the provider terraform schema?

  - [ ] Make sure you explain why this option is important to you, why it should be important to everyone. Describe your use-case with detail and provide examples where possible.
  - [ ] If it is a very special case, consider using the _XSLT_ support in the provider to tweak the definition instead of opening an issue
  - [ ] Maintainers do not have expertise in every libvirt setting, so please, describe the feature and how it is used. Link to the appropriate documentation

- [x] Is it a bug or something that does not work as expected? Please make sure you fill the version information below:

### Setup

Ubuntu example from this repo, except manually add the mac address to the ```network_interface``` block.

### Steps to Reproduce Issue

1. Add the mac address line: ```mac = "52:54:00:5a:08:d6"```
2. ```terraform apply```: It works, the mac address takes.
3. change the mac address to something else: ```mac = "02:38:60:94:88:cc"```
4. ```terraform apply```: No effect

If you change the the xml manually it works, if you destroy and recreate the domain it also works. Shutting the domain down before applying doesn't help.

**Comment by MalloZup** [comment]
@brokenpylons hi. Yeah this is a known issue of the terraform provider.

In fact not only `updating` the mac doesn't work, but other operations.

This is due to the fact that the `update` operation in Libvirt is quite complex and for some network or other device, updating is quite disruptive change.

I remember have investigated on this time ago and basically when `updating` a network, you need to reboot all your domains etc or other complicated intruisive operations. 

This is historically why some update operation, mostly all are not supported. for this reason

**Comment by brokenpylons** [comment]
@MalloZup yes, I figured as much, but I actually just wanted to change the mac address of the domain not in the network. I had a problem where the information in the dhcp block in the network was lost at reboot of the server. I tried to work around this issue by creating the network in virsh and then use terraform for the domains, but not being able to change the mac was limiting, so I had to move back to virsh completely.

A solution from my perspective would be to add a ```mac``` option at the top level, which just controls the mac address of the domain and then extend the dhcp section of the network to allow manually adding the entries. For the update maybe you could detach the old interface and attach the new one like described [here](https://aboullaite.me/effectively-restarting-kvm-libvirt-network/).

**Comment by melkypie** [comment]
The same seems to be adding any new network interfaces or switching the network_interface to a different bridge. This pretty crucial for us as we are trying to run networking devices on kvm and will need to change networking interfaces pretty often. It seems like it can be done https://netdevops.me/2017/changing-libvirt-bridge-attachment-in-a-running-domain-aka-on-the-fly/ with virsh update-device.

**Comment by scabala** [comment]
Is it also a problem with latest provider version?

**Comment by dmacvicar** [comment]
ℹ️ ℹ️ ℹ️ ℹ️ ℹ️ ℹ️

This issue report is relevant to the legacy version of the provider, which is now in the v0.8 branch of this repository.

Future development is based on a new provider, which is not compatible with this one, nor does share code.

As the new provider solves many issues with the legacy, and to make development in my free time more enjoyable, I have decided to close all bugs for the legacy provider, and to ask to check if the bug would apply to the new one. This also to encourage trying the new version.

- https://registry.terraform.io/providers/dmacvicar/libvirt/latest

and check the documentation:

- https://github.com/dmacvicar/terraform-provider-libvirt/blob/main/README.md
- https://registry.terraform.io/providers/dmacvicar/libvirt/latest/docs

You are free to reopen the bug for v0.8. It may encourage someone to fix it. My efforts will go into the new provider.

I ask you to check the new provider and re-evaluate this bug. 🙏
